bigc/resources/customer_groups.py

```diff
@@ -0,0 +1,35 @@
+from collections.abc import Iterator
+
+from bigc._client import BigCommerceV3APIClient
+
+
+class BigCommerceCustomerGroupsAPI:
+    def __init__(self, store_hash: str, access_token: str):
+        self._v2_client = BigCommerceV2APIClient(store_hash, access_token)
+
+    def all(self) -> Iterator[dict]:
+        """Return an iterator for all customer groups"""
+        return self._v2_client.paginated_request('GET', '/customer_groups')
+
+    def get(self, customer_group_id: int) -> dict:
+        """Get a specific customer group by its ID"""
+        return self._v2_client.request('GET', f'/customer_groups/{customer_group_id}')
+
+    def count(self) -> int:
+        """Get a count of all customer groups"""
+        return self._v2_client.request('GET', f'/customer_groups/count')
+
+    def create(self, *, name: str) -> dict:
+        """Create a customer group"""
+        payload = {
+            'name': name,
```

Comment:
My only suggestion here would be to collect extra kwargs like 

https://github.com/MedShift/bigc/blob/master/bigc/resources/customers.py#L46

```suggestion
            'name': name,
            **kwargs
```